### PR TITLE
STY: F-strings and repr

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [ ] ref #29886 and #29547
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/tseries/offsets/test_offsets.py [hunk 1]
```diff
@@ -4170,9 +4169,9 @@ def _test_offset(self, offset_name, offset_n, tstart, expected_utc_offset):
 
     def _make_timestamp(self, string, hrs_offset, tz):
         if hrs_offset >= 0:
-            offset_string = "{hrs:02d}00".format(hrs=hrs_offset)
+            offset_string = f"{hrs_offset:02}00"
```
Comment (WillAyd) [inline pandas/tests/tseries/offsets/test_offsets.py]: Is the `d` format specifier automatically inferred based on supplied object here? Might be worth double checking this modification

**Review by WillAyd — COMMENTED** [review]
Generally lgtm one question

File: pandas/tests/tseries/offsets/test_offsets.py (hunk 1 above)
Comment (MomIsBestFriend) [reply pandas/tests/tseries/offsets/test_offsets.py]: No it doesn't automatically infer the object unfortunately.

Comment (WillAyd) [reply pandas/tests/tseries/offsets/test_offsets.py]: Sure that or I think you can just keep the same `02d` format spec

**Review by WillAyd — APPROVED** [review]
lgtm

**Comment by jreback** [comment]
thanks @MomIsBestFriend 

if you would add a code-check for using ``r!`` would be great!

**Comment by MomIsBestFriend** [comment]
> thanks @MomIsBestFriend
> 
> if you would add a code-check for using `r!` would be great!

Sure I'll just eliminate every occurrence of ```!r``` first

**Comment by jreback** [comment]
thanks @MomIsBestFriend  :)

**Pull request merged**